### ISSUE TITLE
BufReader Checksum Error

### ISSUE THREAD

**Issue by Leoyzen** [jerry73204/rust-tfrecord#3]
Hello, thanks for your contribution.
I encountered problem when using this repo try reading tfrecord  with check_integrity=true.
```
[2021-06-02 13:52:20][ERROR]idx@62680: UnexpectedEofError
[2021-06-02 13:52:20][ERROR]idx@62681: ChecksumMismatchError { expect: "0x0a75f6ec", found: "0x63c4f097" }
[2021-06-02 13:52:20][ERROR]idx@62682: ChecksumMismatchError { expect: "0x0a0b1263", found: "0x71959e9e" }
[2021-06-02 13:52:20][ERROR]idx@62683: ChecksumMismatchError { expect: "0x1b0a6572", found: "0x801e5dc9" }
[2021-06-02 13:52:20][ERROR]idx@62684: ChecksumMismatchError { expect: "0xd3100a12", found: "0x92913227" }
[2021-06-02 13:52:20][ERROR]idx@62685: ChecksumMismatchError { expect: "0x4635efaf", found: "0xcaa9b1d1" }
[2021-06-02 13:52:20][ERROR]idx@62686: ChecksumMismatchError { expect: "0x67616d69", found: "0x690cd862" }                           
[2021-06-02 13:52:20][ERROR]idx@62687: ChecksumMismatchError { expect: "0xd8ff27a2", found: "0x16c5f22f" }
```
It seems reading len_buf failed when using BufReader and meet the incompetible end of buffer.
After some digging, I found this: https://github.com/rust-lang/rust/issues/22570
So we should change `reader.read` to `reader.read_exact`?

**Comment by jerry73204** [comment]
With regard to #4, I made a fix in recent commit ([link](https://github.com/jerry73204/rust-tfrecord/tree/e16fc05e5798a4fe16bf86c615bbc9b250bdd883)). Can you check if it works for you?

**Comment by jerry73204** [comment]
The issue is fixed and release 0.8.0 is now available. Closing it.
